engine/kv_engine.cpp

```diff
@@ -2032,25 +2147,32 @@ Status KVEngine::listRegisterRecovered() {
 }
 
 template <typename DelayFree>
-Status KVEngine::listDestroy(List* list, DelayFree delay_free) {
+Status KVEngine::listDestroy(List* list, DelayFree delay_free, bool local) {
   // Currently, every list operation locks the whole list
   // and delay_free is not necessary.
   // We use delay_free here so that we can enable some lockless
   // operations in the future.
   while (list->Size() > 0) {
-    auto token = version_controller_.GetLocalSnapshotHolder();
-    list->PopFront(
-        [&](DLRecord* elem) { delay_free(elem, token.Timestamp()); });
+    auto ts = local ? version_controller_.GetLocalSnapshotHolder().Timestamp()
+                    : version_controller_.GetGlobalSnapshotToken()->Timestamp();
+    list->PopFront([&](DLRecord* elem) { delay_free(elem, ts); });
   }
-  auto token = version_controller_.GetLocalSnapshotHolder();
-  list->Destroy([&](DLRecord* lrec) { delay_free(lrec, token.Timestamp()); });
+  auto ts = local ? version_controller_.GetLocalSnapshotHolder().Timestamp()
+                  : version_controller_.GetGlobalSnapshotToken()->Timestamp();
+  list->Destroy([&](DLRecord* lrec) { delay_free(lrec, ts); });
   return Status::Ok;
 }
 
 Status KVEngine::listDestroy(List* list) {
+  {
```

Comment:
This one is just a wrapper. Put the actual logic in `Status KVEngine::listDestroy(List* list, DelayFree delay_free, bool local)`